# 25 jul 2018

In [1]:
import pandas as pd
import requests
from utils.camara_obter_dados import obter_dados
from utils.camara_obter_dados import obter_votos

In [2]:
# Pretty Print
def pp(df):
    for row in df.itertuples():
        print('{} {}/{} - {} ({})\n'.format(row.siglaTipo, row.numero, row.ano, row.ementa, row.id))

In [3]:
# Check se há votações para estas proposições
def check(df):
    for row in df.itertuples():
        url = 'https://dadosabertos.camara.leg.br/api/v2/proposicoes/{}/votacoes'.format(row.id)
        request = requests.get(url)
        if request.status_code == 200:
            response = request.json()
            if (len(response['dados'])):
                print('{} - {} votações'.format(row.id, len(response['dados'])))
        else:
            print('Erro ao tentar carregar dados da API (id: {})'.format(row.id))

## Segurança

### Porte de Armas/Desarmamento

In [37]:
# https://dadosabertos.camara.leg.br/api/v2/proposicoes?itens=100&ordem=ASC&ordenarPor=id&keywords=desarmamento

desarmamento = obter_dados(
    endpoint='/proposicoes',
    filtros=[
        'keywords=desarmamento',
        'siglaTipo=',
        'numero=',
        'ano='
    ]
)

desarmamento

Página 1 de 1

ano                                             ementa       id  idTipo  \
0   2010  Acrescenta o inciso XI ao art. 6º, da Lei nº 1...   469624     139   
1   2012  Disciplina as normas sobre aquisição, posse, p...   541857     139   
2   2015  Altera a Lei nº 10.826, de 22 de dezembro de 2...   946617     139   
3   2015  Proíbe a fabricação e comercialização de banan...   949181     139   
4   2015  Altera a Lei n.º 10.826, de 22 de dezembro de ...  1049116     139   
5   2015  Altera a redação do art. 6º da Lei 10.826, de ...  1585017     139   
6   2016  Altera a Lei nº 10.826, de 22 de dezembro de 2...  2087266     139   
7   2018  Susta os arts. 37 e 48 do Decreto no 5.123, de...  2173377     408   
8   2018  Introduz modificações na legislação penal e pr...  2178170     139   
9   2018  Altera a Lei nº 10.826, de 22 de dezembro de 2...  2179333     139   
10  2018  Altera a Lei nº 7.102, de 20 de junho de 1983,...  2180742     139   
11  2018  Altera a Lei n.º 10.826, de 22 de dezembro de ...  2181133     139   
12  2018  Altera o inciso X do art. 6º da Lei nº 10.826,...  2181722     139   

    numero siglaTipo                                                uri  
0     6971        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
1     3722        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
2      344        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
3      484        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
4      805        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
5     2393        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
6     5505        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
7      921       PDC  https://dadosabertos.camara.leg.br/api/v2/prop...  
8    10372        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
9    10437        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
10   10511        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
11   10539        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
12   10607        PL  https://dadosabertos.camara.leg.br/api/v2/prop...

In [42]:
# for row in desarmamento.itertuples():
#     print('{} {}/{} - {} ({})\n'.format(row.siglaTipo, row.numero, row.ano, row.ementa, row.id))

pp(desarmamento)

PL 6971/2010 - Acrescenta o inciso XI ao art. 6º, da Lei nº 10.826, de 22 de dezembro de 2003, que dispõe sobre o registro, posse e comercialização de armas de fogo e munição, sobre o Sistema Nacional de Armas - Sinarm, define crimes e dá outras providências. (469624)

PL 3722/2012 - Disciplina as normas sobre aquisição, posse, porte e circulação de armas de fogo e munições, cominando penalidades e dando providências correlatas. (541857)

PL 344/2015 - Altera a Lei nº 10.826, de 22 de dezembro de 2003, que dispõe sobre registro, posse e comercialização de armas de fogo e munição, sobre o Sistema Nacional de Armas - Sinarm, define crimes e dá outras providências.  (946617)

PL 484/2015 - Proíbe a fabricação e comercialização de banana de dinamite e altera a Lei nº 10.826, de 22 de dezembro de 2003 - Estatuto do Desarmamento, para determinar a identificação de explosivos fabricados ou comercializados no país.  (949181)

PL 805/2015 - Altera a Lei n.º 10.826, de 22 de dezembro de 2003, pa

Segundo pesquisas online, a [PL 3722/2012](https://dadosabertos.camara.leg.br/api/v2/proposicoes/541857) é que tem ganhado mais força nos últimos anos, como mostra [esta reportagem](https://exame.abril.com.br/brasil/entenda-o-projeto-de-lei-que-enfraquece-estatuto-do-desarmamento/).

Porém, **ainda não houve nenhuma votação sobre este Projeto de Lei**.

In [43]:
check(desarmamento)

### Redução da Maioridade Penal

In [49]:
maioridade_penal = obter_dados(
    endpoint='/proposicoes',
    filtros=[
        'keywords=maioridade',
        'siglaTipo=',
        'numero=',
        'ano='
    ]
)

maioridade_penal

Página 1 de 1

ano                                             ementa       id  idTipo  \
0  2002  Acrescenta §§ aos arts. 104 e 105 da Lei nº 8....    68352     139   
1  2016  Torna obrigatório o registro de identificação ...  2087763     139   
2  2016  Estabelece a data em que a vítima completar de...  2121185     139   

   numero siglaTipo                                                uri  
0    7197        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
1    5537        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
2    6690        PL  https://dadosabertos.camara.leg.br/api/v2/prop...

In [51]:
pp(maioridade_penal)

PL 7197/2002 - Acrescenta §§ aos arts. 104 e 105 da Lei nº 8.069, de 13 de julho de 1990, que dispõe sobre o Estatuto da Criança e do Adolescente e dá outras providências, para permitir a aplicação de medidas sócio-educativas aos infratores que atingirem a maioridade penal. (68352)

PL 5537/2016 - Torna obrigatório o registro de identificação do comprador de produtos derivados de tabaco ou fumo.  (2087763)

PL 6690/2016 - Estabelece a data em que a vítima completar dezoito anos como termo inicial do prazo prescricional dos crimes cometidos contra crianças e adolescentes. (2121185)



In [52]:
check(maioridade_penal)

## Educação

### Reforma do Ensino Médio

In [53]:
# MPV 746/16
ensino_medio = obter_dados(
    endpoint='/proposicoes',
    filtros=[
        'keywords=',
        'siglaTipo=MPV',
        'numero=746',
        'ano=2016'
    ]
)

ensino_medio

Página 1 de 1

ano                                             ementa       id  idTipo  \
0  2016  Institui a Política de Fomento à Implementação...  2112490     291   

   numero siglaTipo                                                uri  
0     746       MPV  https://dadosabertos.camara.leg.br/api/v2/prop...

In [54]:
pp(ensino_medio)

MPV 746/2016 - Institui a Política de Fomento à Implementação de Escolas de Ensino Médio em Tempo Integral, altera a Lei nº 9.394, de 20 de dezembro de 1996, que estabelece as diretrizes e bases da educação nacional, e a Lei nº 11.494 de 20 de junho 2007, que regulamenta o Fundo de Manutenção e Desenvolvimento da Educação Básica e de Valorização dos Profissionais da Educação, e dá outras providências.  NOVA EMENTA: Altera as Leis nºs 9.394, de 20 de dezembro de 1996, que estabelece as diretrizes e bases da educação nacional, e 11.494, de 20 de junho 2007, que regulamenta o Fundo de Manutenção e Desenvolvimento da Educação Básica e de Valorização dos Profissionais da Educação, a Consolidação das Leis do Trabalho - CLT, aprovada pelo Decreto-Lei nº 5.452, de 1º de maio de 1943, e o Decreto-Lei nº 236, de 28 de fevereiro de 1967; revoga a Lei nº 11.161, de 5 de agosto de 2005; e institui a Política de Fomento à Implementação de Escolas de Ensino Médio em Tempo Integral. (2112490)



In [55]:
check(ensino_medio)

2112490 - 10 votações


A votação que "bateu o martelo" e definiu a mudança foi a https://dadosabertos.camara.leg.br/api/v2/votacoes/7333, de acordo com o número de votos indicados pela [Gazeta do Povo](https://www.gazetadopovo.com.br/educacao/reforma-do-ensino-medio-e-aprovada-na-camara-destaques-ficam-para-terca-0vmgfhcmpo1kr3eolxxyp3vlc)

In [6]:
ensino_medio_votos = obter_votos(idVotacao='7333')

ensino_medio_votos

Página 6 de 6

idParlamentar          nomeParlamentar  voto
0           178957        ABEL MESQUITA JR.   Sim
1           178864           ADAIL CARNEIRO   Sim
2           178914     ADALBERTO CAVALCANTI   Sim
3           178890     ADELMO CARNEIRO LEÃO   Não
4           178968          ADELSON BARRETO  null
5           133374            ADEMIR CAMILO   Sim
6           178903         ADILTON SACHETTI   Sim
7           141372           AELTON FREITAS  null
8           160508          AFONSO FLORENCE   Não
9           136811              AFONSO HAMM   Sim
10          178835             AFONSO MOTTA  null
11          160527        AGUINALDO RIBEIRO   Sim
12          178836                ALAN RICK  null
13          160582            ALBERTO FILHO  null
14           73579            ALBERTO FRAGA   Sim
15          160559            ALCEU MOREIRA   Sim
16          160511         ALESSANDRO MOLON   Não
17           73458            ALEX CANZIANI   Sim
18          178972             ALEX MANENTE  null
19          178875          ALEXANDRE BALDY   Sim
20          160545          ALEXANDRE LEITE   Sim
21          178833      ALEXANDRE SERFIOTIS   Sim
22          178936          ALEXANDRE VALLE  null
23          141454           ALFREDO KAEFER   Sim
24          178846       ALFREDO NASCIMENTO  null
25           74057           ALICE PORTUGAL   Não
26          178927            ALIEL MACHADO   Não
27          178937           ALTINEU CÔRTES   Sim
28          178881           ALUISIO MENDES   Sim
29          178973             ANA PERUGINI   Não
..             ...                      ...   ...
483          74376            VANDER LOUBET   Não
484         141553         VANDERLEI MACRIS   Sim
485         178913  VENEZIANO VITAL DO RÊGO  null
486          74308           VICENTE ARRUDA  null
487         160622          VICENTE CANDIDO   Não
488          74283               VICENTINHO   Não
489         137070        VICENTINHO JÚNIOR   Sim
490         178888            VICTOR MENDES   Não
491         141555        VINICIUS CARVALHO   Sim
492         160591          VINICIUS GURGEL  null
493         178992              VITOR LIPPI   Sim
494         178869              VITOR VALIM   Sim
495         160569         WALDENOR PEREIRA   Não
496         141558          WALDIR MARANHÃO   Sim
497         160631             WALNEY ROCHA   Sim
498         178952             WALTER ALVES   Sim
499         160620          WASHINGTON REIS   Sim
500         160518            WELITON PRADO   Não
501          74043       WELLINGTON ROBERTO  null
502         167493           WEVERTON ROCHA   Não
503         160636             WILSON FILHO   Sim
504          73930           WLADIMIR COSTA   Sim
505          74439           WOLNEY QUEIROZ   Não
506         178889                ZÉ CARLOS   Não
507          73931               ZÉ GERALDO   Não
508         160632                 ZÉ SILVA   Sim
509         178923          ZECA CAVALCANTI   Sim
510         160592              ZECA DIRCEU   Não
511         178902               ZECA DO PT  null
512         178949             ZENAIDE MAIA  null

[513 rows x 3 columns]

### Cobrança de Mensalidade para Universidades Públicas

In [67]:
mensalidade_ensino_publico = obter_dados(
    endpoint='/proposicoes',
    filtros=[
        'keywords=educação superior',
        'siglaTipo=',
        'numero=',
        'ano='
    ]
)

mensalidade_ensino_publico

Página 1 de 1

ano                                             ementa       id  idTipo  \
0   2008  Dispõe sobre os valores da Gratificação Tempor...   404688     139   
1   2010  Dispõe sobre o intercâmbio acadêmico de estuda...   465986     139   
2   2011  Regula o exercício da profissão de paisagista ...   515917     139   
3   2012  Autoriza o Poder Executivo a instituir o adici...   560392     139   
4   2012  Altera a Lei nº 9.250, de 26 de dezembro de 19...   562640     139   
5   2013  Dispõe sobre a existência de repositórios  dig...   599605     139   
6   2015  Altera a Lei nº 11.096, de 13 de janeiro de 20...   964530     139   
7   2015  Institui o estágio de estudantes de direito na...   996855     139   
8   2015  Altera o art. 2º da Lei nº 12.591, de 18 de ja...  1594768     139   
9   2015  Institui a Política de Assistência Estudantil ...  2024314     139   
10  2015  Propõe que a Comissão de Defesa do Consumidor ...  2053129     138   
11  2015  Obriga beneficiário de bolsa de estudo de prog...  2055300     139   
12  2017  Altera dispositivo da Lei nº 4.024, de 20 de d...  2165897     139   
13  2018  Propõe que a Comissão de Fiscalização Financei...  2177176     138   
14  2018  Institui o Programa Universidade Operária Bras...  2180925     139   

    numero siglaTipo                                                uri  
0     3724        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
1     6785        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
2     2043        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
3     4742        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
4     4827        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
5     6702        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
6      601        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
7      716        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
8     2478        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
9     3375        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
10      64       PFC  https://dadosabertos.camara.leg.br/api/v2/prop...  
11    3632        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
12    9308        PL  https://dadosabertos.camara.leg.br/api/v2/prop...  
13     177       PFC  https://dadosabertos.camara.leg.br/api/v2/prop...  
14   10523        PL  https://dadosabertos.camara.leg.br/api/v2/prop...

In [68]:
pp(mensalidade_ensino_publico)

PL 3724/2008 - Dispõe sobre os valores da Gratificação Temporária para o Magistério Superior - GTMS, devida aos titulares dos cargos integrantes da Carreira do Magistério Superior de que trata a Lei nº 7.596, de 10 de abril de 1987, lotados ou em exercício nas Instituições Federais de Ensino Superior, vinculadas ao Ministério da Educação ou ao Ministério da Defesa. (404688)

PL 6785/2010 - Dispõe sobre o intercâmbio acadêmico de estudantes de graduação e de pós-graduação no País. (465986)

PL 2043/2011 - Regula o exercício da profissão de paisagista e dá outras providências.  (515917)

PL 4742/2012 - Autoriza o Poder Executivo a instituir o adicional por atividade de risco para os vigilantes de instituições federais de educação superior e de pesquisa científica e tecnológica. (560392)

PL 4827/2012 - Altera a Lei nº 9.250, de 26 de dezembro de 1995, para permitir a dedução, da base de cálculo do imposto de renda da pessoa física, das doações a instituições públicas de educação básica e

In [69]:
mensalidade_ensino_publico = obter_dados(
    endpoint='/proposicoes',
    filtros=[
        'keywords=ensino público',
        'siglaTipo=PEC',
        'numero=395',
        'ano=2014'
    ]
)

mensalidade_ensino_publico

Página 1 de 1

ano                                             ementa      id  idTipo  \
0  2014  Altera a redação do inciso IV do art. 206 da C...  611966     136   

   numero siglaTipo                                                uri  
0     395       PEC  https://dadosabertos.camara.leg.br/api/v2/prop...

Podemos perceber que a API da câmara possui algumas falhas. Por exemplo, ao tentar buscar a [PEC 395/2014](https://dadosabertos.camara.leg.br/api/v2/proposicoes/611966), que _Altera a redação do inciso IV do art. 206 da Constituição Federal, referente à gratuidade do ensino público em estabelecimentos oficiais._, pelas keywords que esta possui, não obtivemos sucesso. Tivemos que buscar em [portais de notícias](http://www2.camara.leg.br/camaranoticias/noticias/EDUCACAO-E-CULTURA/528130-CAMARA-REJEITA-PEC-QUE-AUTORIZAVA-UNIVERSIDADE-PUBLICA-A-COBRAR-POR-CURSO-LATO-SENSU.html) qual foi a real proposição que discutiu a cobrança de mensalidade em universidades públicas para cursos de extensão e pós-graduação lato sensu (especializações).

Foram 304 votos favoráveis e 139 contrários, mas eram necessários 308 votos “sim” para aprovar a proposta, que agora será arquivada.

In [70]:
check(mensalidade_ensino_publico)

611966 - 4 votações


In [4]:
votos = obter_votos(idVotacao='7448')

Página 6 de 6

In [5]:
votos

idParlamentar       nomeParlamentar  voto
0           178957     ABEL MESQUITA JR.   Não
1           178864        ADAIL CARNEIRO   Sim
2           178914  ADALBERTO CAVALCANTI   Sim
3           178890  ADELMO CARNEIRO LEÃO   Não
4           178968       ADELSON BARRETO   Não
5           133374         ADEMIR CAMILO   Sim
6           196359        ADÉRMIS MARINI   Sim
7           178903      ADILTON SACHETTI   Sim
8           141372        AELTON FREITAS   Sim
9           160508       AFONSO FLORENCE   Não
10          136811           AFONSO HAMM   Sim
11          178835          AFONSO MOTTA   Sim
12          160527     AGUINALDO RIBEIRO  null
13          178836             ALAN RICK   Não
14          160582         ALBERTO FILHO  null
15           73579         ALBERTO FRAGA   Sim
16          160559         ALCEU MOREIRA   Sim
17          160511      ALESSANDRO MOLON   Não
18           73458         ALEX CANZIANI   Sim
19          178972          ALEX MANENTE   Sim
20          178875       ALEXANDRE BALDY   Sim
21          160545       ALEXANDRE LEITE   Sim
22          178833   ALEXANDRE SERFIOTIS   Não
23          178936       ALEXANDRE VALLE   Sim
24          141454        ALFREDO KAEFER   Sim
25          178846    ALFREDO NASCIMENTO   Sim
26           74057        ALICE PORTUGAL   Não
27          178927         ALIEL MACHADO   Sim
28          178937        ALTINEU CÔRTES   Sim
29          178881        ALUISIO MENDES   Sim
..             ...                   ...   ...
483         160622       VICENTE CANDIDO   Sim
484          74283            VICENTINHO   Não
485         137070     VICENTINHO JÚNIOR   Não
486         178888         VICTOR MENDES   Sim
487         141555     VINICIUS CARVALHO   Sim
488         160591       VINICIUS GURGEL  null
489         178992           VITOR LIPPI   Sim
490         178869           VITOR VALIM  null
491         189593          WADIH DAMOUS   Não
492         160569      WALDENOR PEREIRA   Não
493         141558       WALDIR MARANHÃO   Não
494         160631          WALNEY ROCHA   Sim
495         178952          WALTER ALVES   Sim
496         141560         WALTER IHOSHI   Sim
497         160518         WELITON PRADO   Não
498          74043    WELLINGTON ROBERTO   Não
499         167493        WEVERTON ROCHA  null
500         193066        WILSON BESERRA   Sim
501         160636          WILSON FILHO   Sim
502          73930        WLADIMIR COSTA   Sim
503          74439        WOLNEY QUEIROZ  null
504          73992          YEDA CRUSIUS  null
505         191945      ZÉ AUGUSTO NALIN   Sim
506         178889             ZÉ CARLOS   Não
507          73931            ZÉ GERALDO   Não
508         160632              ZÉ SILVA   Sim
509         178923       ZECA CAVALCANTI   Não
510         160592           ZECA DIRCEU   Sim
511         178902            ZECA DO PT   Não
512         178949          ZENAIDE MAIA   Não

[513 rows x 3 columns]